src/Conversion/ONNXToKrnl/Math/Reduction.cpp

```diff
@@ -92,6 +132,17 @@ struct ScalarOp<ONNXReduceMeanOp> {
 //===----------------------------------------------------------------------===//
 // Scalar unary ops for lowering ONNXReduceMaxOp
 //===----------------------------------------------------------------------===//
+template <>
```

Comment:
We could keep common code together.
```C++
static Value emitScalarOpForONNXReduceMaxOps (ConversionPatternRewriter &rewriter,
     Location loc, Operation *op, Type elementType,
     ArrayRef<Value> scalarOperands) {
   MathBuilder createMath(rewriter, loc);
   Value lhs = scalarOperands[0];
   Value rhs = scalarOperands[1];
   Value max = createMath.sgt(lhs, rhs);
   return createMath.select(max, lhs, rhs);
 }
```
and then call this static op in the templated `ONNXReduceMaxOp` and `ONNXReduceMaxV13Op` ops. Same for the other ops that uses the same pattern.